CalibCalorimetry/EcalTrivialCondModules/src/EcalLaserCondTools.cc

```diff
@@ -0,0 +1,497 @@
+    }
+
+    //     static const float minp = 0.9;
+    //     static const float maxp = 1.1;
+    static const float minp = 0.00001;
+    static const float maxp = 1.8;
+    //static const float minp = -100;
+    //static const float maxp = 100;
+
+    if (!isfinite(corr.p1) || !isfinite(corr.p2) || !isfinite(corr.p3) || corr.p1 < minp || corr.p1 > maxp ||
+        corr.p2 < minp || corr.p2 > maxp || corr.p3 < minp || corr.p3 > maxp) {
+      fprintf(ferr_, "%d %d %f %f %f\n", t1, (int)detid, corr.p1, corr.p2, corr.p3);
+      corr.p1 = corr.p2 = corr.p3 = 1;
+    }
+
+    if (verb_ > 2) {
+      if (detid.subdetId() == EcalBarrel) {
+        cout << EBDetId(detid);
+      } else if (detid.subdetId() == EcalEndcap) {
+        cout << EEDetId(detid);
+      } else {
+        cout << (int)detid;
+      }
+      cout << ": "
+           << "p1 = " << corr.p1 << "\t"
+           << "p2 = " << corr.p2 << "\t"
+           << "p3 = " << corr.p3 << "\n";
+    }
+    if (detid.subdetId() == EcalBarrel) {
+      if (corr.p1 < 0.55)
+        corr.p1 = .985;
+      if (corr.p2 < 0.55)
+        corr.p2 = .985;
+      if (corr.p3 < 0.55)
+        corr.p3 = .985;
+    }
+    corrSet->setValue((int)detid, corr);
+  }
+
+  try {
+    //Write correction set in DB (one IOV):
+    if (db_->isNewTagRequest("EcalLaserAPDPNRatiosRcd")) {
+      if (verb_)
+        cout << "First IOV, extending starting time." << endl;
+      iovStart = db_->beginOfTime();
+    }
+    timeval t;
+    gettimeofday(&t, 0);
+    if (verb_ > 1)
+      cout << "[" << timeToString(t.tv_sec) << "] "
+           << "Write IOV " << iIov << " starting from " << timeToString(iovStart >> 32) << "... ";
+    db_->writeOne(corrSet, iovStart, "EcalLaserAPDPNRatiosRcd");
+  } catch (const cms::Exception& e) {
+    std::cout << "Failed.\nException cathed while writting to cond DB" << e.what() << "\n";
+  }
+  std::cout << "Suceeded.\n";
+}
+
+bool EcalLaserCondTools::FileReader::nextFile() {
+  for (;;) {
+    ++ifile_;
+    if (ifile_ >= fnames_.size()) {
+      if (verb_ > 1)
+        cout << "No more correction files.\n";
+      return false;
+    }
+    if (verb_ > 1)
+      cout << "Opening file " << fnames_[ifile_] << "\n";
+    f_ = fopen(fnames_[ifile_].c_str(), "r");
+    iline_ = 0;
+    if (f_ == 0) {
+      std::cerr << "Failed to open file " << fnames_[ifile_] << ". File skipped!\n";
+    } else {
+      return true;
+    }
+  }
+}
+
+bool EcalLaserCondTools::FileReader::readTime(int& t1, int t2[92], int& t3) {
+  trim();
+  if ((f_ == 0 || feof(f_)) && !nextFile()) {
+    if (verb_ > 1)
+      cout << "No more record\n";
+    return false;
+  }
+  int i;
+  char* buf = 0;
+  size_t s = 0;
+  while ((i = fgetc(f_)) != 'T' && i != 'L' && i >= 0)
+    getline(&buf, &s, f_);
+  if (buf)
+    free(buf);
+  buf = 0;
+
+  if (i == 'L') {  //last record put 3 consecutive times starting from end of prev. IOV
+    t1 = t3;
+    for (int i = 0; i < 92; ++i)
```

Comment:
I guess this 92 come from `EcalLaserCondTools::fillDb::nLmes`. Then please use a single constant throughout the class.